src/vs/editor/contrib/snippet/snippetController2.ts

```diff
@@ -60,17 +60,17 @@ export class SnippetController2 implements IEditorContribution {
 		return 'snippetController2';
 	}
 
-	insert(
+	async insert(
```

Comment:
🛑 this cannot be async, never. Subsequently all following up async changes cannot happen. 

Making snippets insert async, means that insert suggestion become async and that means you can accept a suggestion, type, and get the suggestion insert after that. This needs a different approach, e.g the snippet completion provider has the change to read the clipboard at the time of computing snippets (async is OK there)